metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart

```diff
@@ -1440,6 +1018,143 @@ void main() {
       },
     );
 
+    test(
+      ".fetchCoverage() throws an ArgumentError if the given build is null",
+      () {
+        final result = adapter.fetchCoverage(null);
+
+        expect(result, throwsArgumentError);
+      },
+    );
+
+    test(
+      ".fetchCoverage() fetches coverage for the given build",
+      () async {
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(defaultArtifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchCoverage() returns null if fetching a workflow run for the given build returns null",
+      () async {
+        whenFetchCoverage(withWorkflowRun: null)
+            .thenSuccessWith(defaultArtifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNull);
+      },
+    );
+
+    test(
+      ".fetchCoverage() returns null if the coverage summary artifact does not exist",
+      () async {
+        const artifactsPage = WorkflowRunArtifactsPage(
+          values: [WorkflowRunArtifact(name: 'test.json')],
+        );
+
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(artifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNull);
+      },
+    );
+
+    test(
+      ".fetchCoverage() does not download any artifacts if the coverage summary artifact does not exist",
+      () async {
+        const artifactsPage = WorkflowRunArtifactsPage(
+          values: [WorkflowRunArtifact(name: 'test.json')],
+        );
+
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(artifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNull);
+        verifyNever(githubActionsClientMock.downloadRunArtifactZip(any));
+      },
+    );
+
+    test(
+      ".fetchCoverage() returns null if an artifact archive does not contain the coverage summary json",
+      () async {
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(defaultArtifactsPage);
+
+        whenDecodeCoverage(withArtifactBytes: coverageBytes).thenReturn(null);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        expect(result, isNull);
+      },
+    );
+
+    test(
+      ".fetchCoverage() fetches coverage using pagination for run artifacts",
+      () async {
+        whenFetchCoverage(withWorkflowRun: defaultWorkflowRun)
+            .thenSuccessWith(artifactsPage);
+
+        when(githubActionsClientMock.fetchRunArtifactsNext(artifactsPage))
+            .thenSuccessWith(defaultArtifactsPage);
+
+        final result = await adapter.fetchCoverage(defaultBuild);
+
+        verify(githubActionsClientMock.fetchRunArtifactsNext(any)).called(1);
+        expect(result, isNotNull);
```

Comment:
```suggestion
        expect(result, isNotNull);
        verify(githubActionsClientMock.fetchRunArtifactsNext(any)).called(1);
```
Should it be concrete value as well instead of `isNotNull`?